### PR TITLE
[DEV-11813] TF support for Resource Owners 

### PR DESCRIPTION
**Pull request by SemyonM-Apono**

### CODE REVIEW

File: internal/models/integration.go [hunk 1]
```diff
@@ -80,9 +94,59 @@ func ConvertToIntegrationModel(ctx context.Context, integration *apono.Integrati
...
 	return &data, nil
 }
 
+func ConvertIntegrationOwnerToData(Owners []aponoapi.IntegrationOwnerTerraform) []IntegrationOwner {
+	if Owners == nil {
+		return nil
+	}
+	var integrationOwners = make([]IntegrationOwner, len(Owners))
+	for i, in := range Owners {
+		var AttributeValue = make([]types.String, len(in.AttributeValue))
```
Comment (tomerstein) [inline internal/models/integration.go:109]:
```suggestion
		var attributeValue = make([]types.String, len(in.AttributeValue))
```
camel case for variable names, please adjust the rest of the code accordingly.

File: internal/models/integration.go [hunk 2]
```diff
@@ -80,9 +94,59 @@ func ConvertToIntegrationModel(ctx context.Context, integration *apono.Integrati
...
+		for j, av := range in.AttributeValue {
```
Comment (tomerstein) [inline internal/models/integration.go:110]: same as before, lets not use index and give more readable names to the iterators

File: internal/models/integration.go [hunk 3]
```diff
@@ -80,9 +94,59 @@ func ConvertToIntegrationModel(ctx context.Context, integration *apono.Integrati
...
+			AttributeValue[j] = basetypes.NewStringValue(av)
+		}
+		var IntegrationId types.String
+		if in.IntegrationId.IsSet() {
+			IntegrationId = basetypes.NewStringValue(*in.IntegrationId.Get())
+		} else {
+			IntegrationId = basetypes.NewStringNull()
+		}
```
Comment (tomerstein) [inline internal/models/integration.go:118]:
```suggestion
		IntegrationId := basetypes.NewStringPointerValue(in.IntegrationId.Get())
```
this code can be simplify with using the pointer to init the variable

File: internal/models/integration.go [hunk 4]
```diff
@@ -80,9 +94,59 @@ func ConvertToIntegrationModel(ctx context.Context, integration *apono.Integrati
...
+		result[i] = ResourceOwnerMapping{
+			TagName:                basetypes.NewStringValue(r.TagName),
+			AttributeType:          basetypes.NewStringValue(r.AttributeType),
+			AttributeIntegrationId: AttributeIntegrationId,
+		}
+	}
+
+	return result
+}
+
```
Comment (tomerstein) [inline internal/models/integration.go:149]: same about arguments and variables in camel case

**Review by tomerstein — CHANGES_REQUESTED** [review]
Good job, lets make few changes

File: internal/mockserver/Integrations.go [hunk 5]
```diff
@@ -16,6 +17,170 @@ var (
...
+
+		resp, err := httpmock.NewJsonResponse(200, integration)
+		if err != nil {
+			return httpmock.NewStringResponse(500, err.Error()), nil
+		}
+
+		return resp, nil
+	})
+
+	httpmock.RegisterResponder(http.MethodDelete, `=~^http://api\.apono\.dev/api/v2/integrations/([^/]+)\z`, func(req *http.Request) (*http.Response, error) {
```
[the remark this answers is not in the document]
Comment (SemyonM-Apono) [reply internal/mockserver/Integrations.go:123]: No it's actually correct. There's no delete endpoint in the new TF API. Upon design step we agreed that we keep delete.

File: internal/mockserver/Integrations.go [hunk 6]
```diff
@@ -16,6 +17,170 @@ var (
...
+
+		resp, err := httpmock.NewJsonResponse(200, messageResponse)
+		if err != nil {
+			return httpmock.NewStringResponse(500, err.Error()), nil
+		}
+
+		return resp, nil
+	})
+
+	httpmock.RegisterResponder(http.MethodGet, `=~^http://api\.apono\.dev/api/v2/integrations-catalog/([^/]+)\z`, func(req *http.Request) (*http.Response, error) {
```
[the remark this answers is not in the document]
Comment (SemyonM-Apono) [reply internal/mockserver/Integrations.go:140]: As agreed - moving to own setup api

File: internal/provider/resource_integration.go [hunk 7]
```diff
@@ -459,55 +460,12 @@ func (r *integrationResource) ValidateConfig(ctx context.Context, req resource.V
...
-	if s.IsNull() {
-		result.Unset()
-	} else {
-		vs := s.ValueString()
-		result.Set(&vs)
+	if model.ResourceOwnerMappings != nil && model.IntegrationOwners == nil {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("resource_owner_mappings"),
+			"Invalid Configuration",
+			"`resource_owner_mappings` cannot be set unless `integration_owners` is also set.",
```
Comment (tomerstein) [inline internal/provider/resource_integration.go:468]: you can define integration owner without resource owners? if so this validation is not good, if not lets change the message to be more like "both field must be set together"

**Review by tomerstein — APPROVED** [review]

**Pull request merged**